src/main/resources/static/assets/js/scripts.js

```diff
@@ -114,7 +119,45 @@ function errorEditCategory(message) {
     $("#updateCategory #errorEditCategory").text(message);
 }
 
-function validateName(inputName) {
-    var regex = /^[A-Za-z0-9_.]{3,20}$/g;
+function switchModal() {
+    document.getElementById('add_item').style.display='none';
+    document.getElementById('add_category').style.display='block';
+}
+
+<!-- Validate Add Item-->
+function validateAddItem() {
+    document.getElementById("errorSaveSelectCategory").style.display = "none";
+    document.getElementById("errorSaveItemName").style.display = "none";
+    document.getElementById("errorSaveItemDate").style.display = "none";
+
+    // Validate if is selected a category
+    var itemCategory = $("#newItem #itemCategory").val();
+    if (itemCategory == '' || itemCategory == null) {
+        document.getElementById("errorSaveSelectCategory").style.display = "block";
+        $("#newItem #errorSaveSelectCategory").text("Please select a category!");
+        return false;
+    }
+
+    // Validate the name
+    var inputName = $("#newItem #itemName").val();
+    if (!validateItemName(inputName)) {
+        document.getElementById("errorSaveItemName").style.display = "block";
+        $("#newItem #errorSaveItemName").text("The name must be between 3 and 20 characters and contain only " +
+            "uppercase or lowercase letters, numbers, spaces or only the '-' or '.' special characters!");
+        return false;
+    }
+
+    // Validate if is selected an expiration date
+    var itemDate = $("#newItem #itemExpirationDate").val();
+    if (itemDate == '' || itemDate == null) {
+        document.getElementById("errorSaveItemDate").style.display = "block";
+        $("#newItem #errorSaveItemDate").text("Please select an experation date!");
+        return false;
+    }
+    return true;
+}
+
+function validateItemName(inputName) {
```

Comment:
you have validateItem name and validateCategory name, and both check the same regex. why not only one function?

Reply:
Because aren't the same. Only for item accept spaces, one accept some special characters... That way I made 2 different validations